### PR TITLE
ocamlPackages.unisim_archisec: init at 0.0.5

### PR DESCRIPTION
**Pull request by HamelinDavid**
###### Description of changes

UNISIM ARCHISEC is a companion project of the binary analysis platform BINSEC. It exposes disassembly metadata and DBA (Dynamic Bitvector Automata) semantics of several instruction set architectures, including ARM and x86.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/unisim_archisec/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildDunePackage, fetchFromGitHub, ... }:
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/unisim_archisec/default.nix:1]: Why the `...`?

File: pkgs/development/ocaml-modules/unisim_archisec/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+buildDunePackage rec {
+  pname = "unisim_archisec";
+  version = "0.0.5";
+
+  src = fetchFromGitHub {
+    owner = "binsec";
+    repo = "unisim_archisec";
+    rev = "${version}";
+    hash = "sha256-ngOSpqQXegN70WxOYXd9p8GmMc9ABVi6RPgas6gy7uk=";
+  };
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/unisim_archisec/default.nix:12]: What about downloading the released source code from there: https://github.com/binsec/unisim_archisec/releases/download/0.0.5/unisim_archisec-0.0.5.tbz

Comment (HamelinDavid) [reply pkgs/development/ocaml-modules/unisim_archisec/default.nix:12]: Sure, I will use `fetchTarball` then

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.unisim_archisec

File: pkgs/development/ocaml-modules/unisim_archisec/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/unisim_archisec/default.nix:10]: Please use `fetchurl` instead.

File: pkgs/development/ocaml-modules/unisim_archisec/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ lib, buildDunePackage }:
+
+buildDunePackage rec {
+  pname = "unisim_archisec";
+  version = "0.0.5";
+
+  src = fetchTarball {
```
Comment (wegank) [inline pkgs/development/ocaml-modules/unisim_archisec/default.nix:7]:
```suggestion
  src = fetchurl {
```

**Pull request merged**